upgrading/glusterfs_upgrades.adoc

```diff
@@ -0,0 +1,85 @@
+[[upgrading-glusterfs-on-openshift]]
+= Upgrading GlusterFS on {product-title}
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[special-considerations-for-glusterfs]]
+== Special Considerations for GlusterFS
+When upgrading GlusterFS on {product-title}, you must upgrade the set of nodes
+where GlusterFS pods are running.
+
+. Manage the daemonset:
++
+----
+$ oc delete ds glusterfs  --cascade=false
+----
+
+. Ensure consistency in the cluster before proceeding:
++
+----
+oc delete pod <gluster_pod>
+----
+
+. Run the following command on each volume to check the self-heal status:
++
+----
+$ gluster volume heal <volname> info
+----
+
+When not upgrading GlusterFS at all in a {product-title} upgrade, `drain` (which
+does not work on daemonsets) and `unschedule` (the daemonset will still create
+or recreate the pod on the node as pending) are run without any consideration
+for what is running there.
+
+There is also the potential for someone to run an upgrade on multiple nodes at
+the same time, which causes issues if more than one are hosting GlusterFS pods.
+
+Even if a serial upgrade is running, there is no guarantee sufficient
+time will be given for node A to complete all of its healing operations before
+node B is terminated. This could leave the cluster in a bad or unknown state.
+Therefore, the following procedure is recommended.
+
+[[upgrading-glusterfs]]
+== Upgrading GlusterFS
+
+. Upgrade the control plane, the master nodes and etcd nodes.
+
+. Upgrade standard infra nodes (router, registry, logging and metrics).
++
+[NOTE]
+====
+If any of the nodes in those groups are running GlusterFS , perform step 4 of
+this procedure at the same time. GlusterFS nodes must be upgraded along with
+other nodes in their class (`app` versus `infra`), one at a time.
+====
+
+. Upgrade standard nodes running application containers.
++
+[NOTE]
+====
+If any of the nodes in those groups are running GlusterFS , perform step 4 of
+this procedure at the same time. GlusterFS nodes must be upgraded along with
+other nodes in their class (`app` versus `infra`), one at a time.
+====
+
+. Upgrade the GlusterFS cluster one node at a time.
+
+.. Remove the daemonset label from the node.
+
+.. Run the upgrade playbook.
+
+.. Relabel the node.
+
+.. Wait for the gluster pod to respawn and appear.
+
+.. `rsh` into the pod and verify all volumes are healed
+
+. Repeat step 4 for for GlusterFS nodes 2 and 3.
```

Comment:
You can probably omit this line, since you already basically say this in line 73.